tests/unit/mathematics/linalg/operations/Viennacl_operations_unittest.cc

```diff
@@ -49,4 +49,23 @@ TEST(LinalgBackendViennaCL, dot)
 	EXPECT_NEAR(result, 5, 1E-15);
 }
 
+TEST(LinalgBackendViennaCL, set_const)
+{
+	sg_linalg->set_gpu_backend(new LinalgBackendViennaCL());
+
+	const index_t size = 5, value = 2;
+	SGVector<int32_t> a(size), b(size);
+	SGVector<int32_t> b_gpu, b_cpu;
+
+	for (index_t i = 0; i < size; ++i)
+		a[i] = value;
+
+	b_gpu = to_gpu(b);
+	set_const(b_gpu, value);
+	b_cpu = from_gpu(b_gpu);
+
+	for (index_t i = 0; i < size; ++i)
+		EXPECT_NEAR(a[i], b_cpu[i], 1E-15);
+}
```

Comment:
This doesn't look very right..
To set a gpu vector, ideally an empty gpu vector with malloced space should be provided.
However, there's no way to initiate such gpu vector from the SGVector interface.

Now, a cpu vector with unknown content is transferred to gpu and the gpu memory is reset by `value`.